src/main/java/org/swisspush/reststorage/RestStorageMod.java

```diff
@@ -18,29 +18,55 @@ public class RestStorageMod extends AbstractVerticle {
     public void start(Promise<Void> promise) {
         ModuleConfiguration modConfig = ModuleConfiguration.fromJsonObject(config());
         log.info("Starting RestStorageMod with configuration: {}", modConfig);
-        Storage storage;
-        switch (modConfig.getStorageType()) {
+
+        createStorage(modConfig).onComplete(event -> {
+            if (event.failed()) {
+                promise.fail(event.cause());
+            } else {
+                Handler<HttpServerRequest> handler = new RestStorageHandler(vertx, log, event.result(), modConfig);
+
+                // in Vert.x 2x 100-continues was activated per default, in vert.x 3x it is off per default.
+                HttpServerOptions options = new HttpServerOptions().setHandle100ContinueAutomatically(true);
```

Comment:
if it was off in 3.x then this results in a behavior change since we are now based on 4.x?

Reply:
In 4.x it is also _off_ as default value

Reply:
Hmm yes, I think we now set the flag to true that was false before.